src/comm/Tcp.cc

```diff
@@ -18,19 +19,16 @@
 #if HAVE_NETINET_IN_H
 #include <netinet/in.h>
 #endif
-#if HAVE_SYS_SOCKET_H
-#include <sys/socket.h>
-#endif
 #include <type_traits>
 
-/// setsockopt(2) wrapper
+/// xsetsockopt(2) wrapper
 template <typename Option>
 static bool
 SetSocketOption(const int fd, const int level, const int optName, const Option &optValue)
 {
     static_assert(std::is_trivially_copyable<Option>::value, "setsockopt() expects POD-like options");
     static_assert(!std::is_same<Option, bool>::value, "setsockopt() uses int to represent boolean options");
-    if (setsockopt(fd, level, optName, reinterpret_cast<const char *>(&optValue), sizeof(optValue)) < 0) {
+    if (xsetsockopt(fd, level, optName, reinterpret_cast<const char *>(&optValue), sizeof(optValue)) < 0) {
```

Comment:
This PR modifies dozens of setsockopt() calls. Nearly all of those calls should be modified again to upgrade callers to use this SetSocketOption() API and its friends. IIRC, a few PRs have already tried to upgrade; they have been unsuccessful, but I am sure it is possible to complete that [2021](https://github.com/squid-cache/squid/commit/db60fdb26c0b92988f7a059655beb051c5b8e545) TODO, eliminating most or even all setsockopt() calls from higher-level Squid code.

@kinkie, do you think eliminating one set of widespread setsockopt() caller changes is a realistic goal worth to spend our resources on? We can discuss specifics if you do.

Reply:
As we discussed yesterday. Yes but not as part of this PR or as a prerequisite for it.
Marking it down for a followup